test/pkg/ptptesthelper/ptptesthelper.go

```diff
@@ -322,3 +322,100 @@ func RebootSlaveNode(fullConfig testconfig.TestConfig) {
 
 	logrus.Info("Rebooting system ends ..............")
 }
+
+// GetPtpPodsPerNode is a helper method to get a map of ptp-related pods (daemonset + operator)
+// that are deployed on each node.
+func GetPtpPodsPerNode() (map[string][]*corev1.Pod, error) {
+	ptpDaemonsetPods, err := client.Client.CoreV1().Pods(pkg.PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: pkg.PtpLinuxDaemonPodsLabel})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get linux-ptp daemonset's pods: %w", err)
+	}
+
+	ptpOperatorPods, err := client.Client.CoreV1().Pods(pkg.PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: pkg.PtPOperatorPodsLabel})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get operator pods: %w", err)
+	}
+
+	// helper list with all ptp pods
+	allPtpPods := []*corev1.Pod{}
+	for i := range ptpDaemonsetPods.Items {
+		allPtpPods = append(allPtpPods, &ptpDaemonsetPods.Items[i])
+	}
+	for i := range ptpOperatorPods.Items {
+		allPtpPods = append(allPtpPods, &ptpOperatorPods.Items[i])
+	}
+
+	podsPerNode := map[string][]*corev1.Pod{}
+	// Fill in the map for the ptp daemon pods.
+	for _, pod := range allPtpPods {
+		if pods, nodeExist := podsPerNode[pod.Spec.NodeName]; nodeExist {
+			pods = append(pods, pod)
+			podsPerNode[pod.Spec.NodeName] = pods
+		} else {
+			podsPerNode[pod.Spec.NodeName] = []*corev1.Pod{pod}
+		}
+	}
+
+	return podsPerNode, nil
+}
+
+// GetPodsTotalCpuUsage uses prometheus metric "container_cpu_usage_seconds_total"
+// to return the total cpu usage by all the given pods.
+// As each query needs to be done inside one of the prometheus pods, an optional
+// param prometheusPod can be set for that purpose. If it's nil, the function
+// will try to get it on every call.
+func GetPodsTotalCpuUsage(pods []*corev1.Pod, prometheusPod *corev1.Pod) (float64, error) {
+	const (
+		// To make sure that prometheus can use rate() with at least two samples,
+		// we should use at least two sampling periods (2 * 30s = 60). We'll add 10
+		// extra seconds as a safeguard.
+		timeWindow = 70 * time.Second
```

Comment:
improvement only(not necesarily in this PR): read scrape_interval from the prometheus container and dynamically adjust the time window.

Reply:
Can we capture these improvement as a issue or task so we don't lose all these good points?